src/arch/arm/machine/gic_v3.c

```diff
@@ -125,11 +125,17 @@ static void gicv3_enable_sre(void)
 {
     uint32_t val = 0;
 
-    /* ICC_SRE_EL1 */
+#ifdef CONFIG_ARM_HYPERVISOR_SUPPORT
+    /* ICC_SRE_EL2/ICC_HSRE */
+    SYSTEM_READ_WORD(ICC_SRE_EL2, val);
+    val |= GICC_SRE_EL2_SRE | GICC_SRE_EL2_EN;
```

Comment:
GICC_SRE_EL2_EN disables trapping to EL2 on change of ICC_SRE_EL1 by EL1, is that what you want? Currently the only field EL1 can change is SRE, the system register access enable bit.

Shouldn't we disable IRQ and FIQ bypass modes and set DIB and DFB to 1? Current hardware has EL3, so these fields are likely read-only versions of ICC_SRE_EL3, but in case that isn't the case, and the hardware supports bypass mode, things may break when enabling it I think. This is true for ICC_SRE_EL1 too.